consensus/src/creation/mod.rs

```diff
@@ -54,41 +68,58 @@ pub struct IO<H: Hasher> {
 async fn create_unit<H: Hasher>(
     round: Round,
     creator: &mut Creator<H>,
-    create_lag: &DelaySchedule,
-    mut can_create: bool,
     incoming_parents: &mut Receiver<Unit<H>>,
-    mut exit: &mut oneshot::Receiver<()>,
-) -> Result<(PreUnit<H>, Vec<H::Hash>), ()> {
-    let mut delay = Delay::new(create_lag(round.into())).fuse();
+) -> Result<(PreUnit<H>, Vec<H::Hash>), CreatorError> {
     loop {
-        if can_create {
-            if let Some(result) = creator.create_unit(round) {
-                return Ok(result);
+        match creator.create_unit(round) {
+            Ok(unit) => return Ok(unit),
+            Err(err) => {
+                debug!(target: "AlephBFT-creator", "Creator unable to create a new unit at round {:?}: {}.", round, err)
```

Comment:
This should be `trace` as it spits one log message per unit, and it's by no means an error. Is it possible to use `Display` for `round` (i.e. `{}` instead of `{:?}`). If yes, please do, because the log is then shorter.

Reply:
done